spec/models/job_posting_spec.rb

```diff
@@ -12,4 +12,36 @@
     it { should belong_to :job_type }
     it { should belong_to :experience_level }
   end
+
+  context ".tag_list" do
+    it "should be unique" do
+      job_posting = create(:job_posting)
+      job_posting.tag_list.add('teste')
+      job_posting.save
+
+      job_posting.tag_list.add('teste')
+      job_posting.save
+
+      expect(job_posting.tag_list.count).to eq 1
+    end
+
+    it "tag is required" do
+      job_posting = create(:job_posting)
+
+      job_posting.tag_list.add('')
+      job_posting.save
+
+      expect(job_posting.tag_list.count).to eq 0
+    end
+
+    it "should have 3 tags maximum" do
+      job_posting = create(:job_posting)
+      job_posting.tag_list = [ 'Ruby', 'Dev Jr', 'Scrum' ]
+      job_posting.save
+
+      job_posting.tag_list.add('teste')
+
+      expect(job_posting).not_to be_valid
```

Comment:
ON this test there's nothing that ensures that the problem is the 4th tag added. Maybe we could use `save!` on line 40. And we could also check the `job_posting.errors`  to make sure we have the validation message related to the maximum tags validation.